CRM/Core/BAO/Address.php

```diff
@@ -493,17 +493,16 @@ public static function &getValues($entityBlock, $microformat = FALSE, $fieldName
         $values['location_type'] = CRM_Utils_Array::value($address->location_type_id, $locationTypes);
       }
       if (!empty($address->state_province_id)) {
-        $address->state = CRM_Core_PseudoConstant::stateProvinceAbbreviation($address->state_province_id, FALSE);
-        $address->state_name = CRM_Core_PseudoConstant::stateProvince($address->state_province_id, FALSE);
+        $values['state_province_abbreviation'] = CRM_Core_PseudoConstant::stateProvinceAbbreviation($address->state_province_id, FALSE);
+        $values['state_province'] = CRM_Core_PseudoConstant::stateProvince($address->state_province_id, FALSE);
       }
```

Comment:
It seems to add a regression on the addDisplay below which expects country / state / world_region to be passed in the object.
```
$address->addDisplay($microformat);
```